src/Interpreters/HashJoin/HashJoin.cpp

```diff
@@ -270,10 +270,17 @@ size_t HashJoin::ScatteredColumns::allocatedBytes() const
 
 size_t HashJoin::NullMapHolder::allocatedBytes() const
 {
+    if (!column)
+        return 0;
     size_t rows = column->size();
     if (rows == 0)
         return 0;
-    return column->allocatedBytes() * columns->selector.size() / rows;
+    // we use cached value. In concurrentHashJoin we can sometimes move the nullmaps to the first shard so it's sometimes not available here
+    size_t sel = selector_rows;
```

Comment:
Maybe we can "cache" it always in the constructor of `NullMapHolder`. This logic looks a little weird.